### DISCUSSION TITLE
lean binary distribution of app (ft. JXL lib)?

### DISCUSSION THREAD

**Discussion by goyalyashpal** [FossifyOrg/General-Discussion#375] [Ideas]
### Checklist

* [x]  I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Gallery/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen) or [closed](https://github.com/FossifyOrg/Gallery/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aclosed) - to which I could contribute my information.
* [x]  I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Gallery/discussions) or [closed](https://github.com/FossifyOrg/Gallery/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
* [x]  I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
* [x]  **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
* [x]  This issue contains only one feature request.
* [x]  I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Gallery/blob/master/CONTRIBUTING.md).
* [ ]  I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

yes, the nearly tripled application size is indeed a concern.  

* https://github.com/FossifyOrg/Gallery/blob/5054b37a96aa1f7d996392d7ec1a9a7155ed0eb0/CHANGELOG.md?plain=1#L14-L15

* > * if application size with JXL library is a concern, I think JXL could be moved to a separately installed plugin. 
  > * I don't know how that's exactly done, I'm afraid, but I've seen some apps doing that.

  _Originally posted by @mgorny in https://github.com/FossifyOrg/Gallery/issues/3#issuecomment-2366736089_

### Why do you want this feature?

* after quick pic app went down, i was now using fossify ...
* on an old "android one" device (NOT to be confused with android v1 lol), ...
* which are very low on resource availability combined with android's stupid restriction on "cant move apps to external sd card" ...
* and 16.5 MiB (v1.1.3) to 42.1 MiB (v1.2.1) is just a bit too much for extremely limited internal storage of these devices

**Comment by OkyDooky** [comment]
It sounds like the multiple apks approach (one per chipset) might be the most convenient, since that trail has been blazed by many other devs and pretty much every app manager/updater (Play Store, F-Droid, Obtainium, etc.) supports it out of the box.
Plus, the app may get larger anyways when other features are added (like restoring panorama support).

**Comment by naveensingh** [comment]
I'm not planning for this now but when this is done, it will be done for all Fossify apps. 

F-Droid metadata will have to be updated, not sure if the switch is easy there.
